services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -311,6 +340,45 @@ async def _start_dynamic_service(
                 wallet_id=project_wallet_id, wallet_name=wallet.name
             )
 
+            # Deal with Pricing plan/unit
+            output = await db.get_project_node_pricing_unit_id(
+                project_uuid=project_uuid, node_uuid=node_uuid
+            )
+            if output:
+                pricing_plan_id, pricing_unit_id = output
+                pricing_unit_get = await rut_api.get_pricing_plan_unit(
+                    request.app, product_name, pricing_plan_id, pricing_unit_id
+                )
+                pricing_unit_cost_id = pricing_unit_get.current_cost_per_unit_id
+                aws_ec2_instances = pricing_unit_get.specific_info["aws_ec2_instances"]
+            else:
+                (
+                    pricing_plan_id,
+                    pricing_unit_id,
+                    pricing_unit_cost_id,
+                    aws_ec2_instances,
+                ) = await _fetch_default_pricing_and_hardware_info(
+                    request.app,
+                    product_name,
+                    service_key,
+                    service_version,
+                    project_uuid,
+                    node_uuid,
+                )
+                await db.connect_pricing_unit_to_project_node(
+                    project_uuid,
+                    node_uuid,
+                    pricing_plan_id,
+                    pricing_unit_id,
+                )
+
+            pricing_info = PricingInfo(
```

Comment:
you can even later use `zip` or slice functions in reverse to slit the NamedTuple into these two.